{
  try {
    out.close();
    part.commit();
  }
 catch (  IOException|RuntimeException e) {
    cancel();
    throw e;
  }
}
